File: Objective-C Example/TheoremReachiOSExampleApp/ViewController.m
//
//  ViewController.m
//  TheoremReachiOSExampleApp
//
//  Created by Tom Hammond on 6/14/17.
//  Copyright © 2017 theoremreach. All rights reserved.
//

#import "ViewController.h"
#import <TheoremReachSDK/TheoremReach.h>

@implementation ViewController

- (IBAction)openSurveyButton: (id)sender {
    [self openRewardsCenter];
}

- (IBAction)openHotSurveyButton: (id)sender {
    [self openHotSurvey:@"0bb25f4d-6579-4218-bb5f-202f6db32972"];
}

- (void)openRewardsCenter {
    if ([[TheoremReach getInstance] isSurveyAvailable]) {
        //placement example for additional targeting
        //        [TheoremReach showRewardCenter:@"be4aa618-3c11-498a-92f9-43bb01f2a4c9"];
        [TheoremReach showRewardCenter];
    }
}

- (void)openHotSurvey: (NSString *)acuid {
    if ([acuid length] < 1) { return; }
    TheoremReach *theoremReach = [TheoremReach getInstance];
    if (theoremReach == nil) { return; }
    [theoremReach showHotSurvey:acuid];
 }

@end

File: Swift Example/TheoremReachiOSExampleApp/ViewController.swift
//
//  ViewController.swift
//  Swift Example App
//
//  Created by Tom Hammond on 4/24/19.
//  Copyright © 2019 theoremreach. All rights reserved.
//

import UIKit
import TheoremReachSDK

class ViewController: UIViewController {
    
    @IBAction func openSurveyButton(_ sender: Any) {
        openRewardsCenter()
    }
    
    @IBAction func openHotSurveyButton(_ sender: Any) {
        openHotSurvey("0bb25f4d-6579-4218-bb5f-202f6db32972")
    }
    
    func openRewardsCenter() {
        guard let theoremReach = TheoremReach.getInstance() else { return }
        if theoremReach.isSurveyAvailable {
            TheoremReach.showRewardCenter()
        }
    }
    
    func openHotSurvey(_ surveyAcuid: String) {
        guard surveyAcuid != "" else { return }
        guard let theoremReach = TheoremReach.getInstance() else { return }
        theoremReach.showHotSurvey(surveyAcuid)
    }

}

